File: src/main/java/com/qulix/shilomy/trainingtask/web/repository/TaskRepository.java
package com.qulix.shilomy.trainingtask.web.repository;

import com.qulix.shilomy.trainingtask.web.entity.TaskEntity;

import java.util.List;

public interface TaskRepository {
    void addTask(TaskEntity task);
    void removeTask(TaskEntity task);
    void updateTask(TaskEntity task);

    List<?> query(TaskSpecification specification);
}

File: src/main/kotlin/com/makshilo/trainingtask/repository/TaskRepository.kt
package com.makshilo.trainingtask.repository

import com.makshilo.trainingtask.model.Task

interface TaskRepository {
  fun add(task: Task): Task
  fun update(task: Task): Task
  fun remove(id: Long)
  fun findById(id: Long): Task?
  fun findAll(): List<Task>
}
